DS_and_Algorithm/Day3/Selectionsort.py

```diff
@@ -0,0 +1,22 @@
+def selection_sort(arr):
+    n = len(arr)
+
+    for i in range(n):
+        # Find the minimum element in the unsorted part
+        min_index = i
+        for j in range(i-1, n):
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Incorrect inner loop range in selection sort</b></div>
<div id="fix">

The range in the inner loop seems incorrect. Using `range(i-1, n)` starts from `i-1` which could lead to negative indices and miss comparing elements. Consider using `range(i+1, n)` instead to compare with unsorted elements.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
        for j in range(i+1, n):
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/2#issuecomment-2536074306>#861357</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged